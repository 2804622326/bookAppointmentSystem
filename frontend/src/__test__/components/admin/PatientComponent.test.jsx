import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import PatientComponent from '../../../components/admin/PatientComponent.jsx';

jest.mock('../../../components/patient/PatientService.js', () => ({
  getPatients: jest.fn(() => Promise.resolve({ data: [] })),
}));

jest.mock('../../../components/common/Paginator.jsx', () => () => <div data-testid="paginator" />);
jest.mock('../../../components/user/UserFilter.jsx', () => () => <div data-testid="user-filter" />);


test('shows no data message when there are no patients', async () => {
  const { asFragment } = render(<PatientComponent />);
  await waitFor(() => screen.getByText(/no.*patients data.*available at the moment/i));
  expect(screen.getByText(/no.*patients data.*available at the moment/i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
