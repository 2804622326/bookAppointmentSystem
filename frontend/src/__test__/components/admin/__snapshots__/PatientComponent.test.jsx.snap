// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows no data message when there are no patients 1`] = `
<DocumentFragment>
  <main>
    <div
      class="text-center mt-5"
    >
      <h5>
        No  patients data  available at the moment
      </h5>
    </div>
  </main>
</DocumentFragment>
`;
